conda/testing/gateways/fixtures.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2012 Anaconda, Inc
+# SPDX-License-Identifier: BSD-3-Clause
+
+import socket
+import tempfile
+from pathlib import Path
+
+import pytest
+
+from .helpers import have_minio
+
+
+def s3_server(xprocess):
+    """
+    Mock a local S3 server using `minio`
+
+    This requires:
+    - pytest-xprocess: runs the background process
+    - minio: the executable must be in PATH
+
+    Note it will be given EMPTY! The test function needs
+    to populate it. You can use
+    `conda.testing.helpers.populate_s3_server` for that.
+    """
+    # The 'name' below will be the name of the S3 bucket containing
+    # keys like `noarch/repodata.json`
+    NAME = "s3_server"
+    PORT = 9000
+
+    from xprocess import ProcessStarter
+
+    temp = tempfile.TemporaryDirectory()
+    (Path(temp.name) / NAME).mkdir()
+
```

Comment:
Let's instantiate the minio instance here,

```suggestion
    minio = Minio()
```